plugins/importexport/native/filter/NativeXmlPublicationFilter.inc.php

```diff
@@ -0,0 +1,272 @@
+	 */
+	function &process(&$document) {
+		$importedObjects =& parent::process($document);
+
+		return $importedObjects;
+	}
+
+	/**
+	 * Populate the submission object from the node, checking first for a valid section and published_date/issue relationship
+	 * @param $publication Publication
+	 * @param $node DOMElement
+	 * @return Publication
+	 */
+	function populateObject($publication, $node) {
+		$deployment = $this->getDeployment();
+		$context = $deployment->getContext();
+
+		$sectionAbbrev = $node->getAttribute('section_ref');
+		if ($sectionAbbrev !== '') {
+			$sectionDao = DAORegistry::getDAO('SectionDAO'); /* @var $sectionDao SectionDAO */
+			$section = $sectionDao->getByAbbrev($sectionAbbrev, $context->getId());
+			if (!$section) {
+				$deployment->addError(ASSOC_TYPE_PUBLICATION, $publication->getId(), __('plugins.importexport.native.error.unknownSection', array('param' => $sectionAbbrev)));
+			} else {
+				$publication->setData('sectionId', $section->getId());
+			}
+		}
+		// check if article is related to an issue, but has no published date
+		// $datePublished = $node->getAttribute('date_published');
+		// $issue = $deployment->getIssue();
+		// $issue_identification = $node->getElementsByTagName('issue_identification');
+		// if (!$datePublished && ($issue || $issue_identification->length)) {
+		// 	$titleNodes = $node->getElementsByTagName('title');
+		// 	$deployment->addError(ASSOC_TYPE_SUBMISSION, $publication->getId(), __('plugins.importexport.native.import.error.publishedDateMissing', array('articleTitle' => $titleNodes->item(0)->textContent)));
+		// }
+
+		$this->populatePublishedPublication($publication, $node);
+
+		return parent::populateObject($publication, $node);
+	}
+
+	/**
+	 * Handle an element whose parent is the submission element.
+	 * @param $n DOMElement
+	 * @param $publication Publication
+	 */
+	function handleChildElement($n, $publication) {
+		switch ($n->tagName) {
+			case 'article_galley':
+				$this->parseArticleGalley($n, $publication);
+				break;
+			case 'issue_identification':
+				// do nothing, because this is done in populatePublishedSubmission
+				break;
+			case 'pages':
+				$publication->setData('pages', $n->textContent);
+				break;
+			case 'covers':
+				import('plugins.importexport.native.filter.NativeFilterHelper');
+				$nativeFilterHelper = new NativeFilterHelper();
+				$nativeFilterHelper->parsePublicationCovers($this, $n, $publication);
+				break;
+			default:
+				parent::handleChildElement($n, $publication);
+		}
+	}
+
+	/**
+	 * Get the import filter for a given element.
+	 * @param $elementName string Name of XML element
+	 * @return Filter
+	 */
+	function getImportFilter($elementName) {
+		$deployment = $this->getDeployment();
+		$submission = $deployment->getSubmission();
+		switch ($elementName) {
+			case 'article_galley':
+				$importClass='ArticleGalley';
+				break;
+			default:
+				$importClass=null; // Suppress scrutinizer warn
```

Comment:
Wouldn't this be caught by XML validation? If so, there's no need to code for it here. Or it could be an `assert`, `fatalError` or `Exception` if it's a code path that should never run.